packages/lib-classifier/src/store/subjects/Subject/TranscriptionReductions/TranscriptionReductions.js

```diff
@@ -57,24 +57,67 @@ const TranscriptionReductions = types
       }
     }
 
+    /**
+    Generate an array of consensus lines for a single frame.
+    */
+    function frameConsensus(line, options) {
+      if (!line.id) {
+        line.id = cuid()
+      }
+      return constructLine(line, options)
+    }
+
+    /**
+    Generate an array of consensus lines for a single reduction and frame.
+    */
+    function reductionConsensus(reduction, frame) {
+      const { parameters } = reduction.data
+      const currentFrameReductions = reduction.data[`frame${frame}`] || []
+      const options = {
+        frame,
+        minimumViews: parameters?.minimum_views || DEFAULT_VIEWS_TO_RETIRE,
+        threshold: parameters?.low_consensus_threshold || DEFAULT_CONSENSUS_THRESHOLD
+      }
+      return currentFrameReductions.map(line => frameConsensus(line, options))
+    }
+
+    /**
+    Unique users from all the lines of a single frame.
+    */
+    function frameUsers(frame) {
+      // A frame is an array of line transcriptions.
+      // Each line has a list of user IDs that contributed to the transcription.
+      const users = frame.map(line => line.user_ids)
+      const uniqueUsers = [...new Set(users.flat())]
+      return uniqueUsers
```

Comment:
Would use same syntax between 91/92 and 107/108. Personally I think it's readable enough to convert the 2 lines into 1 in both cases.